File: client-sdk/csharp/sdk-client/src/VertexCache/SdkClient/ConsoleApp/CliRunner.cs
using System;
using System.Threading.Tasks;
using VertexCache.Sdk.Results;
using VertexCache.SdkClient.Config;
using VertexCache.SdkClient.ConsoleApp;
using VertexCache.Sdk.Core;
using VertexCache.Sdk.Transport;

namespace VertexCache.SdkClient.ConsoleApp
{
    public class CliRunner
    {
        private readonly IVCacheClient _client;
        private readonly VertexCacheSdkOptions _options;

        public CliRunner(IVCacheClient client, VertexCacheSdkOptions options)
        {
            _client = client;
            _options = options;
        }

        public async Task RunInteractiveAsync()
        {
            ConsoleDisplay.PrintWelcomeBanner(_options);

            while (true)
            {
                ConsoleDisplay.PrintPrompt(_options);

                string? input = System.Console.ReadLine();
                if (string.IsNullOrWhiteSpace(input)) continue;

                if (input.Trim().Equals("exit", StringComparison.OrdinalIgnoreCase))
                    break;

                var (cmd, args) = CommandParser.Parse(input);

                var result = await _client.RunCommandAsync(cmd, args);

                if (result.IsSuccess)
                {
                    System.Console.WriteLine(result.Message); // ✅ Print success message
                }
                else
                {
                    System.Console.WriteLine($"[ERROR] {result.Message}"); // ✅ Print error
                }
            }
        }
    }
}

File: client-sdk/go/sdk-client/console_app/cli_runner.go
package console_app

import (
	"bufio"
	"fmt"
	"os"
	"sdk/core"
	"strings"
)

type CliRunner struct {
	client  core.VCacheClient
	options *core.VertexCacheSdkOptions
}

func NewCliRunner(client core.VCacheClient, options *core.VertexCacheSdkOptions) *CliRunner {
	return &CliRunner{
		client:  client,
		options: options,
	}
}

func (r *CliRunner) RunInteractiveAsync() {
	PrintWelcomeBanner(r.options)

	scanner := bufio.NewScanner(os.Stdin)
	for {
		PrintPrompt(r.options)

		if !scanner.Scan() {
			break
		}

		input := scanner.Text()
		if strings.TrimSpace(input) == "" {
			continue
		}

		if strings.EqualFold(strings.TrimSpace(input), "exit") {
			break
		}

		cmd, args := ParseCommand(input)
		result := r.client.RunCommand(cmd, args)

		if result.IsSuccess() {
			fmt.Println(result.Message())
		} else {
			fmt.Printf("[ERROR] %s\n", result.Message())
		}
	}
}
